client/starwhale/api/_impl/instance.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+from http import HTTPStatus
+from urllib.parse import urlparse
+
+import requests
+
+from starwhale.utils import fmt_http_server
+from starwhale.consts import HTTPMethod, UserRoleType, STANDALONE_INSTANCE
+from starwhale.base.cloud import CloudRequestMixed
+from starwhale.utils.config import SWCliConfigMixed
+from starwhale.base.uri.instance import Instance
+
+DEFAULT_HTTP_TIMEOUT = 90
+
+
+def login(
+    instance: str,
+    alias: str = "",
+    username: str = "",
+    password: str = "",
+    token: str = "",
+) -> None:
+    """Login to a Starwhale Server instance or Cloud instance.
+
+    Provide either username + password or token to login. Login information will be saved in `~/.config/starwhale/config.yaml`.
+    login function is equivalent to `swcli instance login` command.
+
+    We cannot login to a Standalone instance.
```

Comment:
Also add a comment that users could only have one logined instance at a time. That is a concern for multi-instance operating.